pkgs/swift2objc/lib/src/transformer/_core/dependencies.dart

```diff
@@ -0,0 +1,198 @@
+import '../../ast/_core/interfaces/declaration.dart';
+import '../../ast/_core/interfaces/enum_declaration.dart';
+import '../../ast/_core/interfaces/function_declaration.dart';
+import '../../ast/_core/interfaces/variable_declaration.dart';
+import '../../ast/_core/shared/parameter.dart';
+import '../../ast/_core/shared/referred_type.dart';
+import '../../ast/declarations/compounds/class_declaration.dart';
+import '../../ast/declarations/compounds/members/initializer_declaration.dart';
+import '../../ast/declarations/compounds/protocol_declaration.dart';
+import '../../ast/declarations/compounds/struct_declaration.dart';
+
+/// Gets the type name from a string type by removing other characters like 
+Set<String> _getTypeNames(String type) {
+  // Remove optional markers (?) and square brackets ([])
+  type = type.replaceAll(RegExp(r'\?|!|[\[\]]'), '');
+
+  // Remove annotations (words starting with @)
+  type = type.replaceAll(RegExp(r'@\w+'), '');
+
+  // Extract unique type names using regex
+  final matches = RegExp(r'\b\w+\b').allMatches(type);
+
+  // Return unique type names as a set
+  return matches.map((match) => match.group(0)!).toSet();
+}
+
+// TODO: Type restrictions have not yet been implemented in system
```

Comment:
TODOs should link to a bug. If there isn't one, file one first, then link to the bug like `TODO(https://github.com/dart-lang/native/issues/...): Type restrictions...`

Same below